### PR TITLE
Support IBM cloud as backend

### LINKED ISSUE #469: Fairing supports IBM Public Cloud as backend
/kind feature

**Describe the solution you'd like**
Fairing already support GKE, AWS and Azure, should also support IBM Public Cloud as backend

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by jinchihe**
**What this PR does / why we need it**:
Support ibm cloud 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #469 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/479)

### CODE REVIEW

**Comment by jinchihe** [comment]
/assign @Tomcli 

Would you please take a look? thanks @Tomcli 
By the way, I will update mnist example for IBM Cloud once the PR merged.

/cc @animeshsingh

File: kubeflow/fairing/cloud/ibm_cloud.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        self.client = ibm_boto3.client(
+            "s3",
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_accesss_key,
+            endpoint_url=cos_endpoint_url
+        )
+
+    def create_bucket(self, bucket_name):
+        """
+        Create Bucket in IBM IBM Cloud Object Storage.
```
Comment (Tomcli) [inline kubeflow/fairing/cloud/ibm_cloud.py]:
```suggestion
        Create Bucket in IBM Cloud Object Storage.
```

File: kubeflow/fairing/cloud/ibm_cloud.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        :param bucket_name(str): Bucket name.
+        """
+        try:
+            self.client.head_bucket(Bucket=bucket_name)
+        except ClientError:
+            self.client.create_bucket(Bucket=bucket_name)
+
+    def upload_to_bucket(self, blob_name, bucket_name, file_to_upload):
+        """
+        Uploaded file to IBM IBM Cloud Object Storage.
```
Comment (Tomcli) [inline kubeflow/fairing/cloud/ibm_cloud.py]:
```suggestion
        Uploaded file to IBM Cloud Object Storage.
```

File: kubeflow/fairing/cloud/ibm_cloud.py (hunk 1 above)
Comment (jinchihe) [reply kubeflow/fairing/cloud/ibm_cloud.py]: Updated, thanks.

File: kubeflow/fairing/cloud/ibm_cloud.py (hunk 2 above)
Comment (jinchihe) [reply kubeflow/fairing/cloud/ibm_cloud.py]: Updated, thanks.

**Comment by jinchihe** [comment]
@Tomcli Thanks for the reviewing. Addressed your comments. 
OK, will do that once get chance. Thanks.

**Pull request merged**